Modules/DSCParser/Modules/DSCParser.psm1

```diff
@@ -80,17 +84,70 @@ function ConvertFrom-CIMInstanceToHashtable
         $statements = $ChildObject.PipelineElements.Expression.SubExpression.Statements
         foreach ($statement in $statements)
         {
-            $result += ConvertFrom-CIMInstanceToHashtable -ChildObject $statement
+            $result += ConvertFrom-CIMInstanceToHashtable -ChildObject $statement `
+                                                          -ResourceName $ResourceName
         }
     }
     else
     {
         $result = @{}
         $KeyPairs = $ChildObject.CommandElements[2].KeyValuePairs
         $CIMInstanceName = $ChildObject.CommandElements[0].Value
+
+        # Get the CimClass associated with the current CimInstanceName
+        $CIMClassObject = Get-CimClass -ClassName $CimInstanceName `
+                                       -Namespace 'ROOT/Microsoft/Windows/DesiredStateConfiguration' `
+                                       -ErrorAction SilentlyContinue
+
+        if ($null -eq $CIMClassObject)
+        {
+            $dscResourceInfo = Get-DSCResource -Name $ResourceName
+            $InvokeParams = @{
+                Name        = $ResourceName
+                Method      = 'Get'
+                Property    = @{
+                    'dummyValue' = 'dummyValue'
+                }
+                ModuleName  = @{
+                    ModuleName    = $dscResourceInfo.ModuleName
+                    ModuleVersion = $dscResourceInfo.Version
+                }
+                ErrorAction = 'Stop'
+            }
+
+            try
+            {
+                Invoke-DscResource @InvokeParams
```

Comment:
`Invoke-DscResource` will always fail. Which is expected. But the following code is not reachable and should be placed outside this try-catch-block.

Reply:
Not sure I follow. Why would you say the code is never reachable? This seems to work correctly.

Reply:
I'm also having trouble running this code since it always fails with the below, I tried it by defining these 2 vars as follows.

```powershell
$ResourceName = "IntuneAppConfigurationPolicy"
$CimInstanceName = "MSFT_DeviceManagementConfigurationPolicyAssignments"
```

```powershell
Undefined property dummyValue
At line:8, char:2
Buffer: 
irectResourceAccess";
};^                                                                                                                                                                                                                insta
At line:1 char:1
+ Invoke-DscResource @InvokeParams
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : SyntaxError: (root/Microsoft/...gurationManager:String) [], CimException
+ FullyQualifiedErrorId : MiClientApiError_Failed
+PSComputerName        : localhost
```

Reply:
This is expected. We are calling this function with non valid values. This will immediately terminate the function call with @ricmestre error message.

Reply:
Ok, I think I get it now. So essentially, going with:

Invoke-DscResource @InvokeParams -ErrorAction SilentlyContinue

Would fix the flow

Reply:
It would fix the flow whereas it will not throw an exception if the resource cannot be found.

Reply:
I don't think the issue is the resource not being found as much as dummyvalue not being an accepted parameter on the resource. I am running a few more tests now.

Reply:
That would do the trick to reach the code beneath, but I just tested a complex resource and not all values are extracted, I'll report at the end.

Reply:
> I don't think the issue is the resource not being found as much as dummyvalue not being an accepted parameter on the resource. I am running a few more tests now.

This would be another exception like within the if in the catch block.